content/hacking-atom/sections/hacking-on-atom-core.md

```diff
@@ -0,0 +1,307 @@
+
+{{/windows}}
+
+{{#linux}}
+
+###### TypeError: Unable to watch path
+
+If you get following error with a big traceback right after Atom starts:
+
+```
+TypeError: Unable to watch path
+```
+
+you have to increase number of watched files by inotify.  For testing if this is the reason for this error you can execute:
+
+``` command-line
+$ sudo sysctl fs.inotify.max_user_watches=32768
+```
+
+then restart Atom.  If Atom now works fine, you can make this setting permanent:
+
+``` command-line
+$ echo 32768 | sudo tee -a /proc/sys/fs/inotify/max_user_watches
+```
+
+See also [#2082](https://github.com/atom/atom/issues/2082).
+
+###### /usr/bin/env: node: No such file or directory
+
+If you get this notice when attempting to run any script, you either do not have
+Node.js installed, or node isn't identified as Node.js on your machine. If it's
+the latter, this might be caused by installing Node.js via the distro package
+manager and not nvm, so entering `sudo ln -s /usr/bin/nodejs /usr/bin/node` into
+your terminal may fix the issue. On some variants (mostly Debian based distros)
+you can use `update-alternatives` too:
+
+``` command-line
+$ sudo update-alternatives --install /usr/bin/node node /usr/bin/nodejs 1 --slave /usr/bin/js js /usr/bin/nodejs
+```
+
+###### AttributeError: 'module' object has no attribute 'script_main'
+
+If you get following error with a big traceback while building Atom:
+
+```
+sys.exit(gyp.script_main()) AttributeError: 'module' object has no attribute 'script_main' gyp ERR!
+```
+
+you need to uninstall the system version of gyp.
+
+On Fedora you would do the following:
+
+``` command-line
+$ sudo yum remove gyp
+```
+
+###### Linux build error reports
+
+Use [this search](https://github.com/atom/atom/search?q=label%3Abuild-error+label%3Alinux&type=Issues)
+to get a list of reports about build errors on Linux.
+
+{{/linux}}
+
+#### Running in Development Mode
+
+Once you have a local copy of Atom cloned and built, you can then run Atom in Development Mode. But first, you have to set the `ATOM_DEV_RESOURCE_PATH` environment variable. It defaults to <span class="platform-mac platform-linux">`~/github`</span><span class="platform-windows">`%USERPROFILE%\github`</span> which then expects the `atom/atom` repository to be cloned into <span class="platform-mac platform-linux">`$ATOM_DEV_RESOURCE_PATH/atom`</span><span class="platform-windows">`%ATOM_DEV_RESOURCE_PATH%\atom`</span>. To run Atom in Dev Mode, use the `--dev` parameter from the terminal:
+
+``` command-line
+$ atom --dev <em>path-to-open</em>
+```
+
+There are a couple benefits of running Atom in Dev Mode:
+
+1. When the `ATOM_DEV_RESOURCE_PATH` environment variable is set correctly, Atom is run using the source code from your local `atom/atom` repository. This means that you don't have to run `script/build` every time you change code. Just restart Atom 👍
```

Comment:
Should we say `script\build` for Windows?

Reply:
Fixed.